web/src/test/auto/headless/engine/keyboard/lazySpan.js

```diff
@@ -0,0 +1,84 @@
+import { assert } from 'chai';
+import fs from 'fs';
+
+import { LazySpan } from 'keyman/engine/keyboard';
+
+const ldmlFixture = fs.readFileSync('../resources/fixtures/imperial_aramaic.kmx', 'binary');
+
+const simpleString = "the quick brown fox jumped over the lazy dog";
+
+const SINGLE_CHAR_RANGE = 256;
+const ENCODED_NUM_BASE = 0;
+
+// Little-endian, and we have to rebuild the number byte-by-byte.
+export function decodeNumber(str, start, end) {
+  end ??= str.length;
+  let num = 0;
+
+  for(let i = end - 1; i >= start; i--) {
+    let val = str.charCodeAt(i);
+    num = num * SINGLE_CHAR_RANGE + val - ENCODED_NUM_BASE;
+  }
+
+  return num;
+}
```

Comment:
Originally from #11088 / `epic/trie-encoding`, but with the endianness flipped and different range / base values.

For a source reference...

https://github.com/keymanapp/keyman/blob/636a3ee05c66a55238700c4a985455088418ece7/common/models/templates/src/trie-compression.ts#L20